### DISCUSSION TITLE
How to use cargo-lambda to build multiple functions

### DISCUSSION THREAD

**Discussion by fabien-h** [cargo-lambda/cargo-lambda#356] [Q&A] [answered]
Hi,

I'm not sure to understand how to use cargo lambda to build multiple functions.

Should it be a workspace where every function is a folder with a cargo.toml and a folder with a lib that is shared between the functions ?

How can `cargo lambda watch` watch multiple functions ?

**Comment by fabien-h** [comment]
So, I dug into the lib to understand how it works and it appears that this works locally:

```
src
    lambdas
        lambda1.rs
        lambda2.rs
        ...
    Cargo.toml
```

And in the `Cargo.toml`:

```
[[bin]]
name = "lambda1"
path = "src/lambdas/lambda1.rs"
test = false

[[bin]]
name = "lambda2"
path = "src/lambdas/lambda2.rs"
test = false

...
```

But I'm not sure this is the way to go and not sure how this will work once deployed.

**Comment by fabien-h** [comment, marked as answer]
omg, I missed the commands part of the documentation...
https://www.cargo-lambda.info/commands/build.html

I found the link in the comments of the code, here : https://github.com/cargo-lambda/cargo-lambda/blob/main/crates/cargo-lambda-deploy/src/lib.rs#L46

I feel stupid now 😅

**Reply by calavera** [reply]
no worries. I'm glad you figured it out. Thanks for giving this tool a try!
